graylog2-web-interface/src/views/logic/queries/SearchType.js

```diff
@@ -0,0 +1,40 @@
+// @flow strict
+import type MessageSortConfig from 'views/logic/searchtypes/messages/MessageSortConfig';
+import type SortConfig from 'views/logic/aggregationbuilder/SortConfig';
+import type { Decorator } from 'views/components/messagelist/decorators/Types';
+import type { Interval } from 'views/components/aggregationbuilder/pivottypes/Interval';
+import type { ElasticsearchQueryString, TimeRange } from './Query';
+
+type SearchTypePivot = {
+  type: string,
+  field: string,
+  limit?: number,
+  interval?: Interval,
+};
```

Comment:
I thought about using a union type here as well, but it would be good to know if the already have a type definition for pivots? I found some, but they were not suitable for a search type pivot.